airflow/jobs/scheduler_job.py

```diff
@@ -880,13 +880,22 @@ def _run_scheduler_loop(self) -> None:
                     # is finished to avoid concurrent access to the DB.
                     self.log.debug("Waiting for processors to finish since we're using sqlite")
                     self.processor_agent.wait_until_finished()
-
-                with create_session() as session:
-                    num_queued_tis = self._do_scheduling(session)
-
-                    self.executor.heartbeat()
-                    session.expunge_all()
-                    num_finished_events = self._process_executor_events(session=session)
+                
+                for attempt in run_with_db_retries():
+                    with attempt:
+                        start_time = time.time()
+                        try:
+                            with create_session() as session:
+                                num_queued_tis = self._do_scheduling(session)
+
+                                self.executor.heartbeat()
+                                session.expunge_all()
+                                num_finished_events = self._process_executor_events(session=session)
+                        except OperationalError as e:
+                            total_time = time.time() - start_time
+                            self.log.error("got a DB exception (retry:%d, total time in seconds: %d), details: %s", attempt.retry_state.attempt_number, total_time, e)
```

Comment:
IIRC tenacity logs the exception automatically, so this is sort of duplicating. We can likely remove `details: %s` from the message.

Also it’d be best for a log message to be a complete sentence with proper capitalisation and punctuation.

Reply:
Do you prefer I completely remove the try & ctch and let tenacity handle it completely (it does print the exception, it just might be a little less easier to infer the context of the error)?

Reply:
Or we could raise a different exception with a clearer message here for tenecity to log.